src/Psalm/Type.php

```diff
@@ -540,25 +540,19 @@ public static function combineUnionTypes(
     /**
      * Combines two union types into one via an intersection
      *
-     *
+     * Returns null (not TNever) if the intersection is empty
      */
     public static function intersectUnionTypes(
-        ?Union $type_1,
-        ?Union $type_2,
+        Union $type_1,
+        Union $type_2,
```

Comment:
Do we *really need* this BC break here? This is a part of our API surface, so the earliest it can change is in Psalm 6.